### PR TITLE
New: Added support for newznab indexers using tvdbid for searching.

### PR DESCRIPTION
**Pull request by Taloth**
This code assumes that nn+ will adopt our supportedParams proposal.

Also updated the ParserService to use tvdbid in ReleaseInfo.
I didn't remove tvrageid, there will be plenty of services still using it, especially for old stuff.

Ended up being a bit bigger than I intended, but when you add ReleaseInfo.TvdbId you get a lot of tests where it 'should' be added.

I didn't want to drop TorznabSettings due to the validator and the fact that we may add params in the future, but I decided to base it on NewznabSettings.

### CODE REVIEW

File: src/NzbDrone.Core/Indexers/Newznab/NewznabRequestGenerator.cs [hunk 1]
```diff
@@ -9,21 +9,79 @@ namespace NzbDrone.Core.Indexers.Newznab
...
+        }
+
+        private bool SupportsTvdbSearch
+        {
+            get
+            {
+                var capabilities = _capabilitiesProvider.GetCapabilities(Settings);
+
+                return capabilities.SupportedTvSearchParameters != null
+                    && capabilities.SupportedTvSearchParameters.Contains("tvdbid")
```
Comment (markus101) [inline src/NzbDrone.Core/Indexers/Newznab/NewznabRequestGenerator.cs]: Trailing `&&` instead? (I think thats what we typically do).

File: src/NzbDrone.Core/Indexers/Torznab/TorznabRssParser.cs [hunk 2]
```diff
@@ -99,6 +100,19 @@ protected override string GetDownloadUrl(XElement item)
             return url;
         }
 
+        protected virtual Int32 GetTvdbId(XElement item)
```
Comment (markus101) [inline src/NzbDrone.Core/Indexers/Torznab/TorznabRssParser.cs]: Saw some other changes to `int` from `Int32` so here too :smile:

**Comment by markus101** [comment]
Didn't notice anything blocking, but I'm curious what the plan is if supportedParams isn't added (on all sites) will we use title searches on the sites that don't add it?

**Comment by Taloth** [comment]
We would have no way to identify em without a manual blacklist.

File: src/NzbDrone.Core/Indexers/Torznab/TorznabRssParser.cs (hunk 2 above)
Comment (Taloth) [inline src/NzbDrone.Core/Indexers/Torznab/TorznabRssParser.cs]: Not in this file, I tend to keep a file consistent until we can get Roslyn to replace em all.

File: src/NzbDrone.Core/Indexers/Newznab/NewznabRequestGenerator.cs (hunk 1 above)
Comment (Taloth) [inline src/NzbDrone.Core/Indexers/Newznab/NewznabRequestGenerator.cs]: It's what we typically do, but in this case leading && seems clearer.
(Btw this was copied from TorznabRequestGenerator, not new code).

**Comment by markus101** [comment]
Nothing jumped out at me, good call on the new classes, cleans thing sup nicely.

**Pull request merged**